cpp/src/interop/from_arrow.cu

```diff
@@ -104,16 +104,21 @@ struct dispatch_to_cudf_column {
     if (array.null_bitmap_data() == nullptr) {
       return std::make_unique<rmm::device_buffer>(0, stream, mr);
     }
-    auto mask = std::make_unique<rmm::device_buffer>(
-      bitmask_allocation_size_bytes(static_cast<size_type>(array.null_bitmap()->size() * CHAR_BIT)),
-      stream,
-      mr);
+    auto null_bitmap_size = array.null_bitmap()->size();
+    auto allocation_size =
+      bitmask_allocation_size_bytes(static_cast<size_type>(null_bitmap_size * CHAR_BIT));
+    auto mask        = std::make_unique<rmm::device_buffer>(allocation_size, stream, mr);
     auto mask_buffer = array.null_bitmap();
     CUDF_CUDA_TRY(cudaMemcpyAsync(mask->data(),
                                   reinterpret_cast<const uint8_t*>(mask_buffer->address()),
-                                  array.null_bitmap()->size(),
+                                  null_bitmap_size,
                                   cudaMemcpyDefault,
                                   stream.value()));
+    auto num_zeros = allocation_size - null_bitmap_size;
+    if (num_zeros > 0) {
+      auto zero_after = static_cast<char*>(mask->data()) + null_bitmap_size;
```

Comment:
I'd like to be more consistent between `char` and `uint8_t` and `std::byte` in libcudf. (e.g. above, the `mask_buffer` is read as `uint8_t*` and this line uses `char*`. We want to move to `std::byte` wherever possible, but I recognize there may be other precedents you're trying to match in this file or elsewhere.

Reply:
Oh no, I just picked the first uint8_t-alike I thought of.

Reply:
Went to `uint8_t *`.